### PR TITLE
Checks are made to see if user is already enrolled and if user has

### PR DESCRIPTION
**Pull request by AndyHWChung**
already gone through the eligibility checks

### CODE REVIEW

**Comment by AndyHWChung** [comment]
https://jira.tools.tax.service.gov.uk/browse/HTS-397

File: app/uk/gov/hmrc/helptosavefrontend/controllers/EligibilityCheckController.scala [hunk 1]
```diff
@@ -18,118 +18,106 @@ package uk.gov.hmrc.helptosavefrontend.controllers
...
-              eligibility.result.fold {
-                auditor.sendEvent(new EligibilityCheckEvent(nino, Some("Unknown eligibility problem")))
-                SeeOther(routes.EligibilityCheckController.notEligible().url)
-              } { _ ⇒
-                auditor.sendEvent(new EligibilityCheckEvent(nino, None))
-                SeeOther(routes.EligibilityCheckController.getIsEligible().url)
-              }
-
-          })
+        checkIfAlreadyEnrolled{ nino ⇒
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/EligibilityCheckController.scala]: format the code please

File: app/uk/gov/hmrc/helptosavefrontend/controllers/EligibilityCheckController.scala [hunk 2]
```diff
@@ -18,118 +18,106 @@ package uk.gov.hmrc.helptosavefrontend.controllers
...
+          checkSession(
+            // there is no session yet
+            performEligibilityChecks(nino, htsContext).fold(
+              e ⇒ handleEligibilityCheckError(e),
+              { case (nino, eligibility) ⇒ handleEligibilityResult(eligibility, nino) }
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/EligibilityCheckController.scala]: `eligibility` can be renamed to `htsSession`

File: app/uk/gov/hmrc/helptosavefrontend/controllers/EnrolmentCheckBehaviour.scala [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import cats.instances.future._
+import play.api.mvc.Result
+import uk.gov.hmrc.helptosavefrontend.enrolment.EnrolmentStore
+import uk.gov.hmrc.helptosavefrontend.models.HtsContext
+import uk.gov.hmrc.helptosavefrontend.services.EnrolmentService
+import uk.gov.hmrc.helptosavefrontend.util.{Logging, NINO}
+import uk.gov.hmrc.play.frontend.controller.FrontendController
+import uk.gov.hmrc.play.http.HeaderCarrier
+
+import scala.concurrent.{ExecutionContext, Future}
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/EnrolmentCheckBehaviour.scala]: unused import

File: app/uk/gov/hmrc/helptosavefrontend/controllers/EnrolmentCheckBehaviour.scala [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      case (nino, EnrolmentStore.Enrolled(itmpHtSFlag)) ⇒
+        // if the user is enrolled but the itmp flag is not set then just
+        // start the process to set the itmp flag here without worrying about the result
+        if (!itmpHtSFlag) {
+          enrolmentService.setITMPFlag(nino).fold(
+            e ⇒ logger.warn(s"Could not start process to set ITMP flag for user $nino: $e"),
+            _ ⇒ logger.info(s"Process started to set ITMP flag for user $nino")
+          )
+        }
+        Future.successful(Ok("You've already got an account - yay!"))
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/EnrolmentCheckBehaviour.scala]: can we use `toFuture` implicit here

File: app/uk/gov/hmrc/helptosavefrontend/controllers/IntroductionController.scala [hunk 5]
```diff
@@ -23,6 +23,8 @@ import play.api.Application
 import play.api.i18n.{I18nSupport, MessagesApi}
 import play.api.mvc._
 import uk.gov.hmrc.helptosavefrontend.views
+import uk.gov.hmrc.helptosavefrontend.util.toFuture
+
 
 import scala.concurrent.Future
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/IntroductionController.scala]: unused import

**Pull request merged**